src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/LicenceOverrideService.kt

```diff
@@ -89,6 +95,46 @@ class LicenceOverrideService(
         eventDescription = reason,
       ),
     )
+
+    // if the request status is to make the licence active - send domain event
+    if (!isCurrentLicenceVariationApproved && newStatus == LicenceStatus.ACTIVE) {
+      domainEventsService.recordDomainEvent(
+        LicenceDomainEventType.LICENCE_ACTIVATED,
+        licence.id.toString(),
+        licence.crn,
+        licence.nomsId,
+      )
+    }
+
+    // if the current licence is a variation approved and the request status is active - send domain event
+    if (isCurrentLicenceVariationApproved && newStatus == LicenceStatus.ACTIVE) {
+      domainEventsService.recordDomainEvent(
+        LicenceDomainEventType.LICENCE_VARIATION_ACTIVATED,
+        licence.id.toString(),
+        licence.crn,
+        licence.nomsId,
+      )
+    }
+
+    // if the current licence is not a variation and the request status is inactivate - send domain event
+    if (licence.kind != LicenceKind.VARIATION && newStatus == LicenceStatus.INACTIVE) {
+      domainEventsService.recordDomainEvent(
+        LicenceDomainEventType.LICENCE_INACTIVATED,
+        licence.id.toString(),
+        licence.crn,
+        licence.nomsId,
+      )
+    }
+
+    // if the current licence is a variation and the request status is inactivate - send domain event
+    if (licence.kind == LicenceKind.VARIATION && newStatus == LicenceStatus.INACTIVE) {
+      domainEventsService.recordDomainEvent(
+        LicenceDomainEventType.LICENCE_VARIATION_INACTIVATED,
```

Comment:
might be better to extract out a `getDomainEventType(licence.kind, newStatus)`?

good opportunity to use `when` and independently testable

Reply:
if there is no event type (`return null`) then you wouldn't need to send an event.

Reply:
this could be simplified further by the suggestion below to move to LicenceKind